### PR TITLE
Mac addr endianess fix

### PR DESCRIPTION
**Pull request by komaljai**
"dstAddr" and "srcAddr" fields are of type "macaddr", which is stored in sk_buff in little endian form.
load_dword will convert big_endian to little_endian and load it to cpu. The issue is ebpf compiler shouldn't call load_dword when there is annotation 'macaddr', instead just load the little-endian format to cpu.

### CODE REVIEW

File: backends/tc/ebpfCodeGen.cpp [hunk 1]
```diff
@@ -408,13 +408,171 @@ void TCIngressPipelinePNA::emitLocalVariables(EBPF::CodeBuilder *builder) {
 }
 
 // =====================EBPFPnaParser=============================
+EBPFPnaParser::EBPFPnaParser(const EBPF::EBPFProgram *program, const IR::ParserBlock *block,
```
Comment (usha1830) [inline backends/tc/ebpfCodeGen.cpp:411]: Please mention the source of this code in comments. as some part is copied from ebpf

**Review by usha1830 — APPROVED** [review]

File: backends/tc/ebpfCodeGen.cpp [hunk 2]
```diff
@@ -408,13 +408,171 @@ void TCIngressPipelinePNA::emitLocalVariables(EBPF::CodeBuilder *builder) {
...
     builder->emitIndent();
     builder->blockStart();
     EBPF::EBPFParser::emit(builder);
     builder->blockEnd(true);
 }
 
+void PnaStateTranslationVisitor::compileExtractField(const IR::Expression *expr,
+                                                     const IR::StructField *field,
+                                                     unsigned alignment, EBPF::EBPFType *type) {
+    unsigned widthToExtract = dynamic_cast<EBPF::IHasWidth *>(type)->widthInBits();
```
Comment (psivanup) [inline backends/tc/ebpfCodeGen.cpp]: Is it possible to get nullptr as result after dynamic_cast? Checking before access the result would be safe. Or assert it.

**Review by psivanup — APPROVED** [review]
Looks fine

File: backends/tc/ebpfCodeGen.cpp (hunk 2 above)
Comment (komaljai) [reply backends/tc/ebpfCodeGen.cpp]: addressed

File: backends/tc/ebpfCodeGen.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/ebpfCodeGen.cpp:411]: addressed

**Pull request closed without merging**